### DISCUSSION TITLE
Welcome to oshi Discussions!

### DISCUSSION THREAD

**Discussion by dbwiddis** [oshi/oshi#1448] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by shvo123** [comment]
Hi,
I would like to know whether their is intention to port OSHI to support AS400 ?

10x,
David

**Reply by dbwiddis** [reply]
To directly answer the question, there is presently no intention to port OSHI.   More broadly, for any new operating system port:
1. It must be supported by JNA, i.e., have a binary built for it.  See [JNA supported operating systems here](https://github.com/java-native-access/jna/tree/master/lib/native).
2. JNA developers need access to a machine to build the binaries noted in paragraph (1).  OSHI developers need access to a machine to develop code and test future changes.
3. If (1) and (2) are met, it's a matter of time, interest, and assistance from the community.
